rotkehlchen/tests/api/test_pnl_csv.py

```diff
@@ -90,94 +92,104 @@ def assert_csv_export_response(response, csv_dir, is_download=False, unicode_che
-    # query it again and make sure that csv is recreated and events are not duplicated
-
-    dbevents.delete_history_events_by_identifier(identifiers=[event_id])
-    query_api_create_and_get_report(
-        server=rotkehlchen_api_server_with_exchanges,
-        start_ts=0,
-        end_ts=1601040361,
-        prepare_mocks=True,
-        unicode_check=True,
-    )
-    response = requests.get(
-        api_url_for(rotkehlchen_api_server_with_exchanges, 'historyexportingresource') +
-        f'?directory_path={csv_dir2}',
-    )
-    assert_csv_export_response(response, csv_dir2, unicode_check=True)
-    # now query the download CSV endpoint
-    response = requests.get(
-        api_url_for(rotkehlchen_api_server_with_exchanges, 'historydownloadingresource'))
-    with tempfile.TemporaryDirectory() as tmpdirname:
-        tempzipfile = Path(tmpdirname, 'temp.zip')
-        extractdir = Path(tmpdirname, 'extractdir')
-        tempzipfile.write_bytes(response.content)
-        with zipfile.ZipFile(tempzipfile, 'r') as zip_ref:
-            zip_ref.extractall(extractdir)
-        assert_csv_export_response(response, extractdir, is_download=True, unicode_check=True)
-    # query it again and make sure that csv is recreated and events are not duplicated
-    query_api_create_and_get_report(
-        server=rotkehlchen_api_server_with_exchanges,
-        start_ts=0,
-        end_ts=1601040361,
-        prepare_mocks=True,
-        unicode_check=True,
-    )
-    response = requests.get(
-        api_url_for(rotkehlchen_api_server_with_exchanges, 'historydownloadingresource'))
-    with tempfile.TemporaryDirectory() as tmpdirname:
-        tempzipfile = Path(tmpdirname, 'temp.zip')
-        extractdir = Path(tmpdirname, 'extractdir')
-        tempzipfile.write_bytes(response.content)
-        with zipfile.ZipFile(tempzipfile, 'r') as zip_ref:
-            zip_ref.extractall(extractdir)
-        assert_csv_export_response(response, extractdir, is_download=True, unicode_check=True)
+        # Query history api to have report data to export
+        query_api_create_and_get_report(
+            server=rotkehlchen_api_server_with_exchanges,
+            start_ts=0,
+            end_ts=1601040361,
+            prepare_mocks=True,
+            unicode_check=True,
+        )
+        csv_dir = str(tmpdir_factory.mktemp('test_csv_dir'))
+        csv_dir2 = str(tmpdir_factory.mktemp('test_csv_dir2'))
+
+        # now query the export endpoint with json body
+        response = requests.get(
+            api_url_for(rotkehlchen_api_server_with_exchanges, 'historyexportingresource'),
+            json={'directory_path': csv_dir},
+        )
+        assert_csv_export_response(response, csv_dir, unicode_check=True)
+        # now query the export endpoint with query params
+        response = requests.get(
+            api_url_for(rotkehlchen_api_server_with_exchanges, 'historyexportingresource') +
+            f'?directory_path={csv_dir2}',
```

Comment:
Since you are modifying this code can you update this to use function params instead of `.get` instead of concatenating the string? I believe the syntax is

```suggestion
            api_url_for(
                rotkehlchen_api_server_with_exchanges,
                'historyexportingresource',
                directory_path=csv_dir2,
            ) 
```

same in other places in the test